Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperation.swift

```diff
@@ -599,122 +847,432 @@ struct RenameCollection: TestOperation {
-    func execute(on target: TestOperationTarget, session: ClientSession?) throws -> TestOperationResult? {
+    func execute(on target: TestOperationTarget, sessionDict _: [String: ClientSession])
+        throws -> TestOperationResult? {
         switch target {
         case let .client(client):
-            _ = try client.watch(session: session)
+            _ = try client.watch(session: nil)
         case let .database(database):
-            _ = try database.watch(session: session)
+            _ = try database.watch(session: nil)
         case let .collection(collection):
-            _ = try collection.watch(session: session)
+            _ = try collection.watch(session: nil)
+        case .session, .testRunner:
+            break
         }
         return nil
     }
 }
 
 struct EstimatedDocumentCount: TestOperation {
-    func execute(on target: TestOperationTarget, session: ClientSession?) throws -> TestOperationResult? {
+    func execute(on target: TestOperationTarget, sessionDict _: [String: ClientSession])
+        throws -> TestOperationResult? {
         guard case let .collection(collection) = target else {
             throw TestError(message: "collection not provided to estimatedDocumentCount")
         }
-        return try .int(collection.estimatedDocumentCount(session: session))
+        return try .int(collection.estimatedDocumentCount(session: nil))
+    }
+}
+
+struct StartTransaction: TestOperation {
+    let options: TransactionOptions
+
+    private enum CodingKeys: CodingKey {
+        case options
+    }
+
+    init() {
+        self.options = TransactionOptions()
+    }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.options = try container.decode(TransactionOptions.self, forKey: .options)
+    }
+
+    func execute(on target: TestOperationTarget, sessionDict _: [String: ClientSession])
+        throws -> TestOperationResult? {
+        guard case let .session(session) = target else {
+            throw TestError(message: "session not provided to startTransaction")
+        }
+        _ = try session.startTransaction(options: self.options)
+        return nil
+    }
+}
+
+struct CommitTransaction: TestOperation {
+    func execute(on target: TestOperationTarget, sessionDict _: [String: ClientSession])
+        throws -> TestOperationResult? {
+        guard case let .session(session) = target else {
+            throw TestError(message: "session not provided to commitTransaction")
+        }
+        _ = try session.commitTransaction()
+        return nil
+    }
+}
+
+struct AbortTransaction: TestOperation {
+    func execute(on target: TestOperationTarget, sessionDict _: [String: ClientSession])
+        throws -> TestOperationResult? {
+        guard case let .session(session) = target else {
+            throw TestError(message: "session not provided to abortTransaction")
+        }
+        _ = try session.abortTransaction()
```

Comment:
this doesnt return anything, so I don't think the `_ =` is needed. ditto for commit transaction.

Reply:
Done.